src/consensus/tx_verify.cpp

```diff
@@ -115,9 +115,27 @@ bool CheckTransaction(const CTransaction& tx, CValidationState& state, bool fCol
         }
     }
 
+    // input scripts cannot have OP_EXCHANGEADDR at all
+    for (const auto &vin: tx.vin) {
+        if (vin.scriptSig.size() >= 1 && vin.scriptSig[0] == OP_EXCHANGEADDR) {
+            return state.DoS(100, false, REJECT_INVALID, "bad-exchange-address-vin");
+        }
+    }
+
+    bool hasExchangeUTXOs = false;
```

Comment:
use `bool hasExchangeUTXOs = tx.HasExchangeAddr()` instead of the if

Reply:
Sounds good